<launch>
	<!-- =================== SET Camera Status (ON/OFF) =================== -->
	<arg name="status" default="True"/>
	<!-- =================== SET Argument =================== -->
	<arg name="camera_id_center" value="2" />
	<arg name="nodelet_manager_name_center" value="top_center" />
	<arg name="camera_name_center" value="/top/center" />
	<!-- =================== SET Argument center CAMERA =================== -->
	<arg name="ext_trigger_mode" value="False" />
	<arg name="color_mode" value="bgr8" />
	<arg name="image_width" value="1152" />
	<arg name="image_height" value="874" />
	<arg name="auto_gain" value="False"/>
	<arg name="master_gain" value="10" />
	<arg name="red_gain" value="100" />
	<arg name="green_gain" value="0" />
	<arg name="blue_gain" value="20" />
	<arg name="auto_exposure" value="False"/>
	<arg name="exposure" value="1"/>
	<arg name="auto_white_balance" value="True"/>
	<arg name="auto_frame_rate" value="True"/>
	<arg name="frame_rate" value="15.0"/>
	<arg name="pixel_clock" value="30"/>
	<arg name="flip_lr" value="False" />
	<!-- =================== center CAMERA =================== -->
	<group if="$(arg status)">
		<node pkg="nodelet" type="nodelet" name="$(arg nodelet_manager_name_center)" args="manager" output="screen" />
		<node pkg="nodelet" type="nodelet" name="ueye_cam_nodelet_$(arg nodelet_manager_name_center)" args="load ueye_cam/ueye_cam_nodelet $(arg nodelet_manager_name_center)">
			<param name="camera_name" type="str" value="$(arg camera_name_center)" />
			<param name="camera_topic" type="str" value="image_raw" />
			<param name="camera_id" type="int" value="$(arg camera_id_center)" />
			<param name="camera_intrinsics_file" type="string" value=" /home/zeabus/catkin_ws/src/src_code/zeabus_vision/zeabus_vision_main/params/top.yaml"/>
			<param name="ext_trigger_mode" type="bool" value="$(arg ext_trigger_mode)" />
			<param name="color_mode" type="str" value="$(arg color_mode)" />
			<param name="image_width" type="int" value="$(arg image_width)" />
			<param name="image_height" type="int" value="$(arg image_height)" />
			<param name="auto_gain" type="bool" value="$(arg auto_gain)" />
			<param name="master_gain" type="int" value="$(arg master_gain)" />
			<param name="red_gain" type="int" value="$(arg red_gain)" />
			<param name="green_gain" type="int" value="$(arg green_gain)" />
			<param name="blue_gain" type="int" value="$(arg blue_gain)" />
			<param name="auto_exposure" type="bool" value="$(arg auto_exposure)" />
			<param name="exposure" type="int" value="$(arg exposure)" />
			<param name="auto_white_balance" type="bool" value="$(arg auto_white_balance)" />
			<param name="auto_frame_rate" type="bool" value="$(arg auto_frame_rate)" />
			<param name="frame_rate" type="double" value="$(arg frame_rate)" />
			<param name="pixel_clock" type="int" value="$(arg pixel_clock)" />
			<param name="flip_lr" type="bool" value="$(arg flip_lr)" />
		</node>
	</group>
	<node pkg="image_proc" name="image_proc" type="image_proc" output="screen" ns="$(arg camera_name_center)"/>
	<arg name="imageType" value="image_rect_color/compressed"/>
	<node pkg="zeabus_vision" name="auto_exposure_top" type="auto_exposure_top.py" output="screen">
		<param name="imageTopicC" type="str" value="/top/center/$(arg imageType)"/>
		<param name="imageClientC" type="str" value="ueye_cam_nodelet_$(arg nodelet_manager_name_center)/"/>
	</node>
</launch>
